vm/devices/vmbus/vmbus_server/src/lib.rs

```diff
@@ -870,13 +874,23 @@ impl ServerTask {
                         .merge(&self.server.with_notifier(&mut self.inner));
                 });
             }
-            VmbusRequest::Save(rpc) => rpc.handle_sync(|()| SavedState {
-                server: self.server.save(),
-            }),
+            VmbusRequest::Save(rpc) => {
+                // TODO: update to true once the save part fix in.
+                let lost_synic_bug_fixed = Some(false);
+                rpc.handle_sync(|()| SavedState {
+                    server: self.server.save(),
+                    lost_synic_bug_fixed,
+                })
+            }
             VmbusRequest::Restore(rpc) => {
+                *IS_LOST_SYNIC_BUG_FIXED.lock() = rpc.0.lost_synic_bug_fixed.unwrap_or(false);
                 rpc.handle_sync(|state| self.server.restore(state.server))
             }
             VmbusRequest::PostRestore(rpc) => {
+                if !*IS_LOST_SYNIC_BUG_FIXED.lock() {
+                    tracing::info!("lost synic bug fix is not in yet, call unstick_channels to mitigate the issue.");
+                    self.unstick_channels(false);
```

Comment:
I'd rather we do this (once) in the `::Start` path below, so that we aren't sending synic messages before `Start`.